### DISCUSSION TITLE
osgEarth - Getting the current width/heigth of the view in degrees

### DISCUSSION THREAD

**Discussion by Warren-Whitfield** [pelicanmapping/osgearth#2465] [Support Q&A] [answered]
Hi everyone
I'm trying to get the width and height of the current view in osgEarth. I need to draw a frame of sorts around the current content. So I need to know the coordinates of positions about 2/3rds to the edge ofthe screen no matter what the zoom level. Any suggestions welcomed please.
Thanks, Warren

**Comment by gwaldron** [comment, marked as answer]
Perhaps you can use this function:
```
MapNode::getGeoPointUnderMouse(view, x, y, output);
```
...where `x` and `y` are screen coordinates. Given two valid queries you can measure the distance between then.

**Reply by Warren-Whitfield** [reply]
Hi Glenn, I can't quite get it to work. The GeoPoint contains 0's regardless of the mouse coords I pass in. I think I may have the incorrect view though, so I'm just checking on that.

**Reply by gwaldron** [reply]
The method is used successfully in several places within osgEarth, so I am confident it is working properly. Let me know if you figure out the issue. Note that the GeoPoint will contain zeros (and be invalid) if the input point does not intersect the map.

**Reply by Warren-Whitfield** [reply]
Not sure what I was doing wrong initially but I've got it working now.
One issue I had was that I was getting the size of the window and using that value directly, forgetting that it should really be size -1 for the last pixel.
Thx for the guidance though!
